File: java/pdfclown.samples.cli/src/org/pdfclown/samples/cli/AcroFormFlatteningSample.java
package org.pdfclown.samples.cli;

import org.pdfclown.documents.Document;
import org.pdfclown.files.File;
import org.pdfclown.tools.FormFlattener;
import org.pdfclown.util.io.IOUtils;

/**
  This sample demonstrates <b>how to flatten the AcroForm fields</b> of a PDF document.

  @author Stefano Chizzolini (http://www.stefanochizzolini.it)
  @since 0.2.0
  @version 0.2.0, 04/08/15
*/
public class AcroFormFlatteningSample
  extends Sample
{
  @Override
  public void run(
    )
  {
    File file = null;
    try
    {
      // 1. Opening the PDF file...
      {
        String filePath = promptFileChoice("Please select a PDF file");
        try
        {file = new File(filePath);}
        catch(Exception e)
        {throw new RuntimeException(filePath + " file access error.",e);}
      }
      Document document = file.getDocument();

      // 2. Flatten the form!
      FormFlattener formFlattener = new FormFlattener();
      formFlattener.flatten(document);
      
      // 3. Serialize the PDF file!
      serialize(file);
    }
    finally
    {
      // 4. Closing the PDF file...
      IOUtils.closeQuietly(file);
    }
  }
}

File: PdfClown-NetStandard/CliSamples/org/pdfclown/samples/cli/AcroFormFlatteningSample.cs
using org.pdfclown.documents;
using org.pdfclown.tools;
using org.pdfclown.files;

namespace org.pdfclown.samples.cli
{
  /**
    <summary>This sample demonstrates how to flatten the AcroForm fields of a PDF document.</summary>
  */
  public class AcroFormFlatteningSample
    : Sample
  {
    public override void Run(
      )
    {
      // 1. Opening the PDF file...
      string filePath = PromptFileChoice("Please select a PDF file");
      using(var file = new File(filePath))
      {
        Document document = file.Document;

        // 2. Flatten the form!
        FormFlattener formFlattener = new FormFlattener();
        formFlattener.Flatten(document);

        // 3. Serialize the PDF file!
        Serialize(file);
      }
    }
  }
}